zulipterminal/core.py

```diff
@@ -253,6 +254,23 @@ def show_edit_history(
             'area:msg'
         )
 
+    def view_in_browser(self, message_id: int) -> None:
+        url = '{}#narrow/near/{}'.format(self.model.server_url, message_id)
+
+        # Don't try to open web browser if running without a GUI
+        if (LINUX and not os.environ.get('DISPLAY')
+                and os.environ.get('TERM')):
+            self.view.set_footer_text(
+                'No DISPLAY environment variable specified. This could '
+                'likely mean your system is running without a GUI.', 3)
+            return
+        # Set a footer text if no runnable browser is located
+        try:
+            webbrowser.get()
+            webbrowser.open(url)
+        except webbrowser.Error as e:
+            self.view.set_footer_text(f"ERROR: {e}", 3)
```

Comment:
Just curious about whether you were able to raise any exception manually to test.

Reply:
Also, let's specify the error context, i.e., the error came while opening the message.

Reply:
Yes, I was able to successfully reproduce this error on my AWS EC2 instance over `ssh`.
The error message is already very specific, so I don't think context would be necessary. See [here](https://stackoverflow.com/questions/39371219/python-webbrowser-check-if-browser-is-available-nothing-happens-when-opening) for the type of error message thrown and how to reproduce it.

Reply:
Thanks for the explanation. Agreed; the error message seems to capture the context itself.